aspnet/migration/http-modules.rst

```diff
@@ -1,8 +1,576 @@
+
+A solution is to branch the pipeline for requests with a given extension, using the ``MapWhen`` extension method.
+This happens in the same ``Configure`` method where you add the other middleware:
+
+.. literalinclude:: http-modules/sample/Asp.Net5/src/Asp.Net5/Startup.cs
+	:language: c#
+	:linenos:
+	:lines: 1-2, 12-15, 46-48, 54-55, 82-87, 105, 109, 110
+	:emphasize-lines: 12-17
+
+``MapWhen`` takes these parameters:
+
+1. A lambda that takes the ``HttpContext`` and returns ``true`` if the request should go down the branch.
+   This means you can branch requests not just based on their extension, but also on request headers, query string parameters, etc.
+
+2. A lambda that takes an ``IApplicationBuilder`` and adds all the middleware for the branch.
+   This means you can add additional middleware to the branch in front of your handler middleware.
+
+The middleware that was added to the pipeline before the branch will be invoked for both requests going down the branch
+and for requests not going down the branch.
+
+Loading middleware options using the options pattern
+----------------------------------------------------
+
+Some modules and handlers have configuration options, which would be stored in the web.config.
+However, in ASP.NET 5 a new configuration model is used in place of web.config.
+
+The new ASP.NET 5 :doc:`configuration system <../fundamentals/configuration>` gives you these options to solve this:
+
+* Use the :ref:`options pattern <options-config-objects>`, as used with MVC. This section shows how.
+* Directly inject the options into the middleware, as shown in the `next section <#loading-middleware-options-through-direct-injection>`_.
+
+1. Create a class to hold your middleware options
+
+    For example:
+
+    .. literalinclude:: http-modules/sample/Asp.Net5/src/Asp.Net5/Middleware/MyMiddlewareWithParams.cs
+        :language: c#
+        :linenos:
+        :lines: 8-12
+        :dedent: 4
+
+2. Store the option values
+
+    The new configuration system allows you to essentially store option values anywhere you want. However, most sites will use
+    *appsettings.json*, so lets use that here too:
+
+    .. literalinclude:: http-modules/sample/Asp.Net5/src/Asp.Net5/appsettings.json
+        :language: json
+        :linenos:
+        :lines: 1, 6-10
+
+    *MyMiddlewareOptionsSection* here is simply a section name. It doesn't have to be the same as the name of your options class.
+
+3. Associate the option values with the options class
+
+    The options pattern uses ASP.NET 5's dependency injection framework to associate the options type (such as ``MyMiddlewareOptions``)
+    with an ``MyMiddlewareOptions`` object that has the actual options.
+
+    To make this work, update your ``Startup`` class:
+
+        a. If you're using *appsettings.json*, make sure it is added to the configuration builder in the ``Startup`` constructor:
```

Comment:
If you're using *appsettings.json*,  add it to the configuration builder in the ``Startup`` constructor: